core/src/com/nuclearthinking/myheroagency/model/Settings.java

```diff
@@ -39,16 +41,35 @@ public void save() {
         Preferences preferences = Gdx.app.getPreferences(PREFERENCES_NAME);
         preferences.clear();
         preferences.putString(LANGUAGE_KEY, language);
+        preferences.putString("height", height);
+        preferences.putString("width", width);
         preferences.flush();
     }
 
-    public String getLanguage() {
+    public static String getLanguage() {
```

Comment:
ну я описал кейс , в каком случае статические переменные помешают

ну и просто лично я стараюсь не прибегать к статическим переменным, если
этого можно избежать

и тут опять же новая точка отказа, обязательно нужно вызвать лоад сеттингс
прежде чем делать статические вызовы переменных, это неочевидный сценарий,
а вот получить экземпляр с помощью лоад сетингс(оно возвращает экземпляр
сеттингов) и у этого экземпляра уже брать параметры, это уже более
привычным кажется, и ожидаемым

вт, 17 мая 2016 г., 7:10 Izonami <notifications@github.com>:

> In core/src/com/nuclearthinking/myheroagency/model/Settings.java
> <https://github.com/nuclearthinking/myheroagency/pull/37#discussion_r63462148>
> :
>
> >          preferences.flush();
> >      }
> >
> > -    public String getLanguage() {
> > +    public static String getLanguage() {
>
> Потому что предполагается что настройки везде будет одинаковыми разве нет
> ?
> А то получится в меню паузы что бы получить актуальные настройки сначала
> надо будет создать экземпляр, потом запустить загрузку, что бы заполнить
> значениями поля и только потом получить значение.
> Вместо того что бы просто сделать вызов нужной настройки и отобразить её
> значение.
>
> Я как то так думал.
> Настройка должна иметь последние актуальные значения, что бы значения были
> актуальными для всех экземпляров класса они должны быть статичными, а раз
> уж они статичные, то почему бы не получить их значения без создания
> экземпляра.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nuclearthinking/myheroagency/pull/37/files/d3e6d14593f67fa7e35391b82e534059989f7113#r63462148>
>